packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -121,6 +129,13 @@ interface ServerlessClusterNewProps {
    */
   readonly scaling?: ServerlessScalingOptions;
 
+  /**
+   * Scaling configuration of an Aurora Serverless V2 database cluster.
+   *
+   * @default - TBC
+   */
+  readonly scalingV2?: ServerlessV2ScalingOptions;
```

Comment:
That `V2` suffix is a bit of an eye sore.

Is there no better way?

Can we perhaps set an enum on the  `ServerlessCluster` class that specifies which version it is, and it then type narrows the interface? Or will it confuse JSII?

```ts
enum ServerlessClusterVersion {
  V1,
  V2,
}

interface ServerlessClusterNewProps {
  version: ServerlessClusterVersion
}

interface ServerlessClusterNewPropsV1 {
  version: ServerlessClusterVersion.V1
  scaling: V1Scaling
}

interface ServerlessClusterNewPropsV2 {
  version: ServerlessClusterVersion.V2
  scaling: V2Scaling
}
```

Reply:
Agree. Actually that was very similar to my previous design. Let me try again.

Reply:
Hi @moltar 

Let's make it a little bit clear.

What I am trying to propose is to reuse the `ServerlessCluster` to create either v1 or v2 clusters. As v1 and v2 have different scaling configuration logic and can't be shared, we would have both scaling interfaces for v1 and v2 in the `ServerlessClusterProps` if we have to reuse the `ServerlessCluster` construct for both scenarios. Your sample seems to create `ServerlessClusterV2` just for v2 in addition to the existing `ServerlessCluster` just for v1. I am fine with that as this is opinionated. Is this something you prefer in the API design?

Reply:
In my example, I am following your proposal to reuse the same class too. I guess my point about that was not very clear.

I was proposing to use [discriminating unions](https://www.typescriptlang.org/docs/handbook/unions-and-intersections.html#discriminating-unions) to type narrow the interface based on the `version` prop.

Reply:
Union types don't work with jsii.